python/tests/test_holy_wvg_bands.py

```diff
@@ -0,0 +1,69 @@
+import unittest
+import meep as mp
+
+
+class TestHolyWvgBands(unittest.TestCase):
+
+    def setUp(self):
+        cell = mp.Vector3(1, 12)
+        b = mp.Block(size=mp.Vector3(1e20, 1.2, 1e20), material=mp.Medium(epsilon=13))
+        c = mp.Cylinder(0.36)
+
+        self.fcen = 0.25
+        self.df = 1.5
+
+        s = mp.Source(
+            src=mp.GaussianSource(self.fcen, fwidth=self.df),
+            component=mp.Hz,
+            center=mp.Vector3(0.1234)
+        )
+
+        sym = mp.Mirror(direction=mp.Y, phase=-1)
+
+        self.sim = mp.Simulation(
+            cell_size=cell,
+            geometry=[b, c],
+            sources=[s],
+            symmetries=[sym],
+            boundary_layers=[mp.Pml(1, direction=mp.Y)],
+            resolution=20
+        )
+
+    def test_run_k_points(self):
+        all_freqs = self.sim.run(mp.interpolate(19, [mp.Vector3(), mp.Vector3(0.5)]), k_points=5)
+
+        self.assertAlmostEqual(all_freqs[17][0].real, 0.1942497850393511)
+        self.assertAlmostEqual(all_freqs[17][0].imag, 0.001381460274205755)
+        self.assertAlmostEqual(all_freqs[18][0].real, 0.19782709203322993)
+        self.assertAlmostEqual(all_freqs[18][0].imag, -0.0013233828667934015)
+        self.assertAlmostEqual(all_freqs[19][0].real, 0.1927618763491877)
+        self.assertAlmostEqual(all_freqs[19][0].imag, 0.001034260690735336)
+        self.assertAlmostEqual(all_freqs[20][0].real, 0.19335527231544278)
+        self.assertAlmostEqual(all_freqs[20][0].imag, 4.6649450258959025e-4)
+
+    def test_fields_at_kx(self):
+        self.sim.k_point = mp.Vector3(3.5)
+        h = mp.Harminv(self.sim, mp.Hz, mp.Vector3(0.1234), self.fcen, self.df)
```

Comment:
I don't understand why the `Harminv` constructor needs to take a simulation argument.   It is passed the simulation object during timestepping, shouldn't that be sufficient?